qt/widgets/instrumentview/src/BankRenderingHelpers.cpp

```diff
@@ -216,7 +216,11 @@ void render2DTexture(const Corners &corners, size_t nX, size_t nY,
   glBegin(GL_QUADS);
 
   // Set the bank normal to facilitate lighting effects
-  setBankNormal(corners.bottomRight(), corners.topLeft(), basePos);
+  try {
+    setBankNormal(corners.bottomRight(), corners.topLeft(), basePos);
```

Comment:
In `Figaro` and `D17` the centres of all detectors lie on a straight line. That's why the normal calculation fails.
My suggestion was to call `setBankNormal` like this:
```
setBankNormal(corners.bottomRight() + bottomRightOffset, corners.topLeft() + topLeftOffset, basePos);
```
that is add offsets from the centres to the detector edges. That way the normal will be calculated and no exception thrown.

Reply:
Thanks for the clarification, I'll apply the changes